provers/sgx/prover/src/lib.rs

```diff
@@ -109,12 +109,11 @@ impl Prover for SgxProver {
             setup(&cur_dir, direct_mode).await?;
         }
 
-        if config.bootstrap {
-            bootstrap(cur_dir.clone(), gramine_cmd()).await?;
-        }
-
-        // Prove: run for each block
-        let sgx_proof = if config.prove {
+        let sgx_proof = if config.bootstrap {
+            bootstrap(cur_dir.clone(), gramine_cmd()).await
+        } else if config.prove {
+            // now prove can not go with bootstrap as we need manually register sgx id.
+            // TODO: auto register sgx id
```

Comment:
Have to be careful for now I guess because this will break proving blocks with prove_block.

Reply:
Yes, will soon get update with the new auto register contract call, then bootstrap can be done right before the prove.
Now for outside docker user, they need to first run boot, then manually register sgx id, then start server daemon. so they are by design separated. Internal users are experienced I guess :)

Reply:
With a 2nd thought, I think better to make it as before, as outside use will not call it, and internal user will get confused why prove not working. Also after next change it will be back as before. 
So, better to revert this else to keep the behavior stable.